### PR TITLE
APB-6203 Granular Permissions support

### PR DESCRIPTION
**Pull request by dg-21-hmrc**

### CODE REVIEW

File: app/uk/gov/hmrc/agentaccesscontrol/service/ESAuthorisationService.scala [hunk 1]
```diff
@@ -170,4 +174,25 @@ class ESAuthorisationService @Inject()(
...
 
+  def checkForRelationship(arn: Arn,
+                           maybeUserId: Option[String],
+                           taxIdentifier: TaxIdentifier)(
+      implicit ec: ExecutionContext,
+      hc: HeaderCarrier): Future[Boolean] =
+    for {
+      // Check whether Granular Permissions are enabled and opted-in for this agent.
+      // If so, when calling agent-client-relationships, we will check whether a relationship exists _for that specific agent user_.
+      // If Granular Permissions are not enabled or opted out, we only check whether a relationship exists with the agency.
```
Comment (geoffreywatson) [inline app/uk/gov/hmrc/agentaccesscontrol/service/ESAuthorisationService.scala:185]: what about the case where the FF is on and the agent has opted in but the client is unassigned to any group? They should get access.

Comment (geoffreywatson) [reply app/uk/gov/hmrc/agentaccesscontrol/service/ESAuthorisationService.scala:185]: (provided the agency is authorised for the client)

Comment (dg-21-hmrc) [reply app/uk/gov/hmrc/agentaccesscontrol/service/ESAuthorisationService.scala:185]: That logic is in agent-client-relationship. In AAC we only check opted in/opted out

Comment (geoffreywatson) [reply app/uk/gov/hmrc/agentaccesscontrol/service/ESAuthorisationService.scala:185]: ok thanks

**Review by geoffreywatson — APPROVED** [review]

**Pull request merged**